src/mol-model-props/common/custom-property.ts

```diff
@@ -34,7 +36,7 @@ namespace CustomProperty {
         readonly getParams: (data: Data) => Params
         readonly defaultParams: Params
         readonly isApplicable: (data: Data) => boolean
-        readonly attach: (ctx: Context, data: Data, props?: Partial<PD.Values<Params>>, addRef?: boolean) => Promise<void>
+        readonly attach: (ctx: Context, data: Data, props?: Partial<PD.Values<Params>>, addRef?: boolean, plugin?: PluginContext) => Promise<void>
```

Comment:
Instead of adding another parameter, I would include PluginContext in Context. 
So we would have:
```
    export interface Context {
        runtime: RuntimeContext
        assetManager: AssetManager
        plugin?: PluginContext
    }
    ...
    readonly attach: (ctx: Context, data: Data, props?: Partial<PD.Values<Params>>, addRef?: boolean) => Promise<void>
```
This applies to both `attach` and `obtain` functions. I think in this way we will keep the code more tidy.

@arose / @dsehnal  please confirm if what I'm saying makes sense.

Reply:
Let's remove parameter altogether, suggesting a different solution.